File: baseLib/src/main/java/com/guider/baselib/utils/UnitUtil.java
package com.guider.baselib.utils;

import android.content.Context;

import com.guider.baselib.device.IUnit;
import com.guider.baselib.device.UnitCN;
import com.guider.baselib.device.UnitImpl;
import com.guider.baselib.device.UnitTW;

import java.util.Locale;

public class UnitUtil {
    public static IUnit getIUnit(Context context) {
        Locale locale = context.getResources().getConfiguration().locale;
        String language = locale.getCountry().toLowerCase();
        if (language.contains("cn"))
            return new UnitImpl(new UnitCN());
        else
            return new UnitImpl(new UnitTW());
    }
}

File: baseLib/src/main/java/com/guider/baselib/utils/UnitUtil.kt
package com.guider.baselib.utils

import com.guider.baselib.device.IUnit
import com.guider.baselib.device.UnitCN
import com.guider.baselib.device.UnitImpl
import com.guider.baselib.device.UnitTW
import com.guider.health.apilib.utils.ApiLibUtil.getCurrentLanguage

object UnitUtil {
    fun getIUnit(): IUnit {
        val language = getCurrentLanguage()
        return if (language == "tw") {
            UnitImpl(UnitTW())
        } else {
            UnitImpl(UnitCN())
        }
    }
}